### PR TITLE
Feature/test cases

### PR DESCRIPTION
**Pull request by ed-rhilbert**
Adding first draft for test cases and a basic routine in notebook only for now to start all test cases on a SchedulerAgent

### CODE REVIEW

**Comment by ed-rhilbert** [comment]
as far as `doc/tutorials/custon_agent.ipynb` is concerned, I think the agents shoud be named consistently, so either 'Dummy' and 'Dummier' or 'Dumb' and 'Dumber'. I like the second option 😁 (not sure you were born ?)

image

File: src/pyosrd/schedules/schedules.py [hunk 1]
```diff
@@ -352,39 +350,79 @@ def has_conflicts(self, train: int) -> bool:
...
             .set_axis(self._df.columns.levels[0], axis=1)
             .astype(float)
         )
 
-    def train_delay(self, train, initial_schedule: 'Schedule') -> pd.DataFrame:
+    def train_delay(self, train, ref_schedule: 'Schedule') -> pd.DataFrame:
+
+        return self.delays(ref_schedule).max().loc[train]
+
+    def compute_weighted_delays(
```
Comment (ed-rhilbert) [inline src/pyosrd/schedules/schedules.py:373]: delay or delays ?
```suggestion
    def compute_weighted_delay(
```

File: src/pyosrd/schedules/schedules.py [hunk 2]
```diff
@@ -352,39 +350,79 @@ def has_conflicts(self, train: int) -> bool:
...
+        compared to the ref_schedule ponderated by weights.
+
+        The formula used is as follow ($s$ are all steps of the schedules,
+        a step being a train in a zone):
+            $$\sum_s [(delayed\_arrival - ref\_arrival)_s \times weight_s]$$
+
+        Parameters
+        ----------
+        self: Schedule
+            The delayed schedule, regulated use to compute the metric
```
Comment (ed-rhilbert) [inline src/pyosrd/schedules/schedules.py:390]: metric --> delays

File: src/pyosrd/schedules/schedules_indicators.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import pandas as pd
+
+from . import Schedule
+
+
```
[the remark this answers is not in the document]
Comment (SimonPierreED) [reply src/pyosrd/schedules/schedules_indicators.py]: fixed in infra_trains

File: src/pyosrd/agents/scheduler_agent.py [hunk 4]
```diff
@@ -112,3 +118,129 @@ def regulated_schedule(self) -> Schedule:
...
+                        regulated_schedule,
+                        weights_.all_steps(sim),
+                    )
+                ]
+            },
+            index=[scenario]
+        )
+
+
+def regulate_scenarii(
```
[the remark this answers is not in the document]
Comment (SimonPierreED) [reply src/pyosrd/agents/scheduler_agent.py]: fixed in infra_trains

File: src/pyosrd/schedules/schedules_indicators.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+def compute_ponderated_delays(
```
[the remark this answers is not in the document]
Comment (SimonPierreED) [reply src/pyosrd/schedules/schedules_indicators.py]: fixed in infra_trains (compute_ponderated_delays has been moved to the schedule class directly)

File: src/pyosrd/agents/scheduler_agent.py [hunk 6]
```diff
@@ -112,3 +118,129 @@ def regulated_schedule(self) -> Schedule:
...
+    scenario 5           98
+    """
+    data = []
+    for test_case in scenarii:
+        data.append(agent.regulate_scenario(test_case))
+
+    return pd.concat(data)
+
+
+def regulate_scenarii_with_agents(
```
[the remark this answers is not in the document]
Comment (SimonPierreED) [reply src/pyosrd/agents/scheduler_agent.py:218]: fixed in infra_trains

File: src/pyosrd/agents/scheduler_agent.py [hunk 7]
```diff
@@ -112,3 +118,129 @@ def regulated_schedule(self) -> Schedule:
...
+                delayed and regulated). Defaults to False.
+
+        Returns:
+            pd.DataFrame: Returns a DataFrame
+            containing the metric for
+            the agent and the given scenario, eg:
+                            agent 1
+            scenario 1           12
+        """
+        module = importlib.import_module(
```
[the remark this answers is not in the document]
Comment (SimonPierreED) [reply src/pyosrd/agents/scheduler_agent.py:153]: fixed in infra_trains

File: src/pyosrd/agents/scheduler_agent.py [hunk 8]
```diff
@@ -112,3 +118,129 @@ def regulated_schedule(self) -> Schedule:
...
+    scenario 4           33         133         233         333       10233
+    scenario 5           98         198         298         398       10298
+
+    """
+    if scenarii == 'all':
+        scenarii = OSRD.scenarii
+    elif scenarii in OSRD.scenarii:
+        scenarii = [scenarii]
+    elif isinstance(scenarii, str):
+        raise ValueError(f"Unknown scenario {scenarii}")
```
[the remark this answers is not in the document]
Comment (SimonPierreED) [reply src/pyosrd/agents/scheduler_agent.py:257]: fixed in infra_trains

File: src/pyosrd/agents/scheduler_agent.py [hunk 9]
```diff
@@ -112,3 +118,129 @@ def regulated_schedule(self) -> Schedule:
...
+        and the given scenarii, eg:
+                    agent 1     agent 2     agent 3     agent 4     agent 5
+    scenario 1           12         112         212         312       10212
+    scenario 2           16         116         216         316       10216
+    scenario 3           22         122         222         322       10222
+    scenario 4           33         133         233         333       10233
+    scenario 5           98         198         298         398       10298
+
+    """
+    if scenarii == 'all':
```
[the remark this answers is not in the document]
Comment (SimonPierreED) [reply src/pyosrd/agents/scheduler_agent.py:252]: fixed in infra_trains

File: src/pyosrd/schedules/schedules.py (hunk 2 above)
Comment (SimonPierreED) [reply src/pyosrd/schedules/schedules.py:390]: fixed in infra_trains

File: src/pyosrd/schedules/schedules.py (hunk 1 above)
Comment (SimonPierreED) [reply src/pyosrd/schedules/schedules.py:373]: fixed in infra_trains

**Pull request merged**